src/multipole.F90

```diff
@@ -0,0 +1,188 @@
+      deallocate(nuc % total)
+      deallocate(nuc % elastic)
+      deallocate(nuc % fission)
+      deallocate(nuc % nu_fission)
+      deallocate(nuc % absorption)
+
+      allocate(nuc % energy(nuc % n_grid))
+      allocate(nuc % total(nuc % n_grid))
+      allocate(nuc % elastic(nuc % n_grid))
+      allocate(nuc % fission(nuc % n_grid))
+      allocate(nuc % nu_fission(nuc % n_grid))
+      allocate(nuc % absorption(nuc % n_grid))
+
+      nuc % total(:) = ZERO
+      nuc % absorption(:) = ZERO
+      nuc % fission(:) = ZERO
+
+      ! Read in new energy axis (converting eV to MeV)
+      call read_dataset(group_id, "energy_points", nuc % energy)
+      nuc % energy = nuc % energy / 1.0e6_8
+
+      ! Get count and list of MT tables
+      call read_dataset(group_id, "MT_count", NMT)
+      allocate(MT(NMT))
+
+      call read_dataset(group_id, "MT_list", MT)
+
+      call close_group(group_id)
+
+      accumulated_fission = .false.
+
+      ! Loop over each MT entry and load it into a reaction.
+      do i = 1, NMT
+        write(MT_n, '(A, I3.3)') '/nuclide/reactions/MT', MT(i)
+
+        group_id = open_group(file_id, MT_n)
+
+        ! Each MT needs to be treated slightly differently.
+        select case (MT(i))
+          case(ELASTIC)
+            call read_dataset(group_id, "MT_sigma", nuc % elastic)
+            nuc % total(:) = nuc % total + nuc % elastic
+          case(N_FISSION)
+            call read_dataset(group_id, "MT_sigma", nuc % fission)
+            nuc % total(:) = nuc % total + nuc % fission
+            nuc % absorption(:) = nuc % absorption + nuc % fission
+            accumulated_fission = .true.
+          case default
+            ! Search through all of our secondary reactions
+            do j = 1, nuc % n_reaction
+              if (nuc % reactions(j) % MT == MT(i)) then
+                ! Match found
+
+                ! Individual Fission components exist, so remove the combined
+                ! fission cross section.
+                if ( (MT(i) == N_F .or. MT(i) == N_NF .or. MT(i) == N_2NF &
+                      .or. MT(i) == N_3NF) .and. accumulated_fission) then
+                  nuc % total(:) = nuc % total - nuc % fission
+                  nuc % absorption(:) = nuc % absorption - nuc % fission
+                  nuc % fission(:) = ZERO
+                  accumulated_fission = .false.
+                end if
+
+                deallocate(nuc % reactions(j) % sigma)
+                allocate(nuc % reactions(j) % sigma(nuc % n_grid))
+
+                call read_dataset(group_id, "MT_sigma", &
+                                  nuc % reactions(j) % sigma)
+                call read_dataset(group_id, "Q_value", &
+                                  nuc % reactions(j) % Q_value)
+                call read_dataset(group_id, "threshold", &
+                                  nuc % reactions(j) % threshold)
+                nuc % reactions(j) % threshold = 1 ! TODO: reconsider implications.
```

Comment:
Just pinging this TODO again so I don't forget about it.  My understanding is that the cross sections in the library have been set to zero where necessary to account for threshold reactions... but I still need to look into this more.